File: apis_py/app/schemas/attempt.py
from pydantic import BaseModel, Field

class AttemptCreate(BaseModel):
    """
    Esquema de creación para la entidad Attempt.

    Atributos:
        participation_id (int): Identificador de la participación. Ejemplo: 1.
        attempt_number (int): Número del intento. Ejemplo: 1.
        type (str): Tipo de intento (Arranque o Envion). Ejemplo: "Arranque".
        weight_lifted (float): Peso levantado en el intento. Ejemplo: 100.0.
        success (bool): Indica si el intento fue exitoso. Ejemplo: True.
    """
    participation_id: int = Field(..., title="ID de la participación", description="Identificador de la participación", example=1)
    attempt_number: int = Field(..., title="Número del intento", description="Número del intento", example=1)
    type: str = Field(..., title="Tipo de intento", description="Tipo de intento (Arranque o Envion)", example="Arranque")
    weight_lifted: float = Field(..., title="Peso levantado", description="Peso levantado en el intento", example=100.0)
    success: bool = Field(..., title="Éxito", description="Indica si el intento fue exitoso", example=True)

class AttemptResponse(BaseModel):
    """
    Esquema de respuesta para la entidad Attempt.
    
    Atributos:
        id (int): Identificador único del intento. Ejemplo: 1.
        participation_id (int): Identificador de la participación. Ejemplo: 1.
        attempt_number (int): Número del intento. Ejemplo: 1.
        type (str): Tipo de intento (Arranque o Envion). Ejemplo: "Arranque".
        weight_lifted (float): Peso levantado en el intento. Ejemplo: 100.0.
        success (bool): Indica si el intento fue exitoso. Ejemplo: True.
        tournament_name (str): Nombre del torneo. Ejemplo: "Torneo Nacional".
        tournament_id (int): Identificador del torneo. Ejemplo: 1.
    """
    id: int = Field(..., title="ID del intento", description="Identificador único del intento", example=1)
    participation_id: int = Field(..., title="ID de la participación", description="Identificador de la participación", example=1)
    attempt_number: int = Field(..., title="Número del intento", description="Número del intento", example=1)
    type: str = Field(..., title="Tipo de intento", description="Tipo de intento (Arranque o Envion)", example="Arranque")
    weight_lifted: float = Field(..., title="Peso levantado", description="Peso levantado en el intento", example=100.0)
    success: bool = Field(..., title="Éxito", description="Indica si el intento fue exitoso", example=True)
    tournament_name: str = Field(..., title="Nombre del torneo", description="Nombre del torneo", example="Torneo Nacional")
    tournament_id: int = Field(..., title="ID del torneo", description="Identificador del torneo", example=1)

    class Config:
        from_attributes = True  # Permite usar atributos de la clase para serialización

File: apis_c#/AthleteApi/AthleteApi/Models/Attempt.cs
using Swashbuckle.AspNetCore.Annotations;

namespace AthleteApi.Models
{
    [SwaggerSchema(Description = "Representa un intento de levantamiento de pesas en un torneo")]
    public class Attempt
    {
        [SwaggerSchema("Identificador único del intento")]
        public int Id { get; set; }

        [SwaggerSchema("Identificador de la participación del atleta en el torneo")]
        public int ParticipationId { get; set; }

        [SwaggerSchema("Número del intento")]
        public int AttemptNumber { get; set; }

        [SwaggerSchema("Tipo de intento (ej. Snatch, Clean and Jerk)")]
        public string? Type { get; set; }

        [SwaggerSchema("Peso levantado en el intento")]
        public double WeightLifted { get; set; }

        [SwaggerSchema("Indica si el intento fue exitoso (1) o fallido (0)")]
        public int Success { get; set; }

        [SwaggerSchema("Nombre del torneo")]
        public string? TournamentName { get; set; }

        [SwaggerSchema("Identificador único del torneo")]
        public int TournamentId { get; set; }
    }
}